docs/cvl/foundry-integration.md

```diff
@@ -0,0 +1,123 @@
+
+While fuzzing is not as complete as formal verification (a fuzzer might “miss” some inputs that would expose a bug), 
+writing fuzz tests via Foundry is often easier for developer since it uses the same Solidity language as the smart contracts themselves.
+Foundry has tooling for writing fuzz tests on smart contracts, see [Foundry Documentation](https://book.getfoundry.sh/forge/fuzz-testing).
+
+This part of the documentation explains how to utilize the Certora Prover to formally verify tests that are written as Foundry fuzz tests.
+
+
+```{caution}
+This feature of the Certora Prover is in alpha state, so issues/unimplemented features are expected. Please contact us if you encounter 
+any issue.
+```
+
+# Usage
+
+There is a minimum of 2 required files to get the Prover to verify Foundry fuzz tests: A `.spec` file (written in CVL), and a `.conf` file. 
+
+- First, we need a `.spec` file written in CVL that will tell the Prover what to verify. The file is very simple, and in the minimal case 
+has exactly one line:
+
+```solidity
+use builtin rule verifyFoundryFuzzTests;
+```
+
+- Second, we need a `.conf` file that will provide the Prover the information of which contract to verify, and what spec file to use. 
+A minimal `.conf` file will look like this:
+
+```json
+{
+  "files": [
+    "path/to/file/with/Foundry/fuzz/tests.sol:<name_of_contract_containing_the_tests>",
+  ],
+  "verify": "name_of_contract_containing_the_tests:path/to/spec/file.spec",
+  "foundry_tests_mode": true,
+}
+```
+
+- Now, to run the tests, execute:
+`certoraRun path/to/conf/file.conf`
+- You will receive a link to a report containing the results of the run.
+
+For a full running example, please also see our [Example Repository](https://github.com/Certora/Examples/).
+
+# Key differences vs. Foundry fuzz testing
+
+In Foundry, fuzz tests start with a blank state as the initial state (i.e. all storage fields are explicitly set to `0`), and one can implement 
+a `setUp()` function in order to bring the state to whatever initial state one wants to run the tests in. In contrast, the Prover starts with 
+an arbitrary initial state and does _not_ assume all storage fields to be initialized with 0 by default. This could cause the Prover to find 
+spurious counter examples. For example, a fuzz test may assumes that a storage value of `balance` is zero for all addresses and the Prover may 
+choose some other initial state violating this basic assumption of the test. 
+
+To restrict the Prover's search space to match the setup of the foundry test, it's possible write a special CVL function  `init_fuzz_tests` 
+that acts as a setup in CVL. 
+
+```solidity
+override function init_fuzz_tests(method f, env e) {
+  // your initial state assumptions here
+}
+ ```
+Depending on your fuzz test, it may be required to use the `reset_storage` command in the `init_fuzz_tests` function. This will explicitly set all 
```

Comment:
```suggestion
Depending on your fuzz test, you may need to use the `reset_storage` command in the `init_fuzz_tests` function. This will explicitly set all 
```